research/sentiment_analysis/sentiment_model.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+def _dynamic_pooling(w_embs):
+    import tensorflow as tf
+    t = tf.expand_dims(w_embs, 2)
+    pool_size = w_embs.shape[1].value / 2
+    pooled = tf.keras.backend.pool2d(t, (pool_size, 1), strides=(pool_size, 1), data_format='channels_last')
+    return tf.squeeze(pooled, 2)
```

Comment:
Could you add some comments for these functions? Like what it does here, and what is the Input and Output. Maybe a class doc will help a lot.

Reply:
ack